### PR TITLE
Capture device events

### PR DESCRIPTION
**Pull request by ut112002**
@VarLad and I have come up with a simple way to capture and use device events in the event loop.

We have also provided a small example 

Kindly take a look and give your opinion

### CODE REVIEW

**Comment by arhik** [comment]
Thanks. It was long overdue change I was hoping to fix.

**Comment by arhik** [comment]
Noticed that camera is not handle properly. I will make quick fix. Camera should actually be used with Orthographic projection instead. For now, I will skip camera events so that your example will run smoothly.

File: examples/uiCircleCaptureExample.jl [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+					flag = true
+				end
+			else
+				if flag
+					popfirst!(renderer.scene.objects)
+					addObject!(renderer, circle, scene.camera)
+					flag = false
+				end 
+			end 
+			@info keyboardState
```
Comment (arhik) [inline examples/uiCircleCaptureExample.jl:56]: You can remove `info`. Its evident visually. I think. Not a big deal though.

Comment (arhik) [reply examples/uiCircleCaptureExample.jl:56]: I pushed an update with info remove. Hope it's fine with you. I also handled camera event issues.

**Comment by arhik** [comment]
There is a way we can avoid regenerating shader code. WorldObject was designed to hide an object with VISIBLE RenderType. Instead of adding and removing an object from scene, which could be expensive, we can instead use VISIBLE RenderType to hide an element we don't want to show.

**Comment by VarLad** [comment]
@arhik Setting `worldobject.rType = VISIBLE` makes th object disappear but, how do we make it appear again?

**Comment by arhik** [comment]
> @arhik Setting `worldobject.rType = VISIBLE` makes th object disappear but, how do we make it appear again? 

It's like xor if I remember correctly.

**Comment by arhik** [comment]
wireFrame.rType ⊻ VISIBLE

**Comment by arhik** [comment]
Precisely like this `wireFrame.rType  = RenderType(wireFrame.rType ⊻ VISIBLE)`

**Comment by arhik** [comment]
Also other way is to create one circle and just change color of the object. Color's uniformData right now is just transformation matrix. There are many good updates in WGPUgfx for transparently setting struct values in julia end and update the WGSLTypes Struct version of the same object. So we need to expose all parameters we can change through uniformData. Camera.jl is perfect reference. Just mentioning if you are interested..

**Pull request merged**